### PR TITLE
fix(nodebuilder/tests): Fix flakeyness in tests with non-empty blocks by waiting til block height 1

### LINKED ISSUE #1669: Swamp tests fail due to rpc error: account not found error
### Celestia Node version

main

### OS

MacOS Monterey 12.6

### Install tools

_No response_

### Others

_No response_

### Steps to reproduce it

1. Run swamp tests
```bash
$ make test-swamp && make test-swamp-race
```

### Expected result

Expecting the rests to pass

### Actual result

Tests fail with:

0.
```
--- FAIL: TestFullReconstructFromBridge (0.99s)
    reconstruct_test.go:73: 
                Error Trace:    /Users/derrandz/Lab/celestia/celestia-node-fork/nodebuilder/tests/reconstruct_test.go:73
                Error:          Received unexpected error:
                                rpc error: code = NotFound desc = account cosmos1pvcm0nqlyyrq0te78qmzdmqhwdegrm8v7hzg5t not found
                Test:           TestFullReconstructFromBridge
```

1.
```
--- FAIL: TestFullReconstructFromLights (1.22s)
    reconstruct_test.go:170: 
                Error Trace:    /Users/derrandz/Lab/celestia/celestia-node-fork/nodebuilder/tests/reconstruct_test.go:170
                Error:          Received unexpected error:
                                rpc error: code = NotFound desc = account cosmos1lcf2vmgrm54c86kxmg6vz6fmjae8dlvnz0fwq2 not found
                Test:           TestFullReconstructFromLights
```

2.
```
--- FAIL: TestSyncLightWithBridge (12.45s)
    sync_test.go:74: 
                Error Trace:    /Users/derrandz/Lab/celestia/celestia-node-fork/nodebuilder/tests/sync_test.go:74
                Error:          Received unexpected error:
                                rpc error: code = NotFound desc = account cosmos1jjhzef6y9cdfh0y0hnhky3l6puqmetgcym0u0c not found
                Test:           TestSyncLightWithBridge
```

3.
```
--- FAIL: TestSyncFullWithBridge (12.67s)
    sync_test.go:187: 
                Error Trace:    /Users/derrandz/Lab/celestia/celestia-node-fork/nodebuilder/tests/sync_test.go:187
                Error:          Received unexpected error:
                                rpc error: code = NotFound desc = account cosmos1y9hm3u984pdtsc2f5udvgj86fusrr3hltcsaur not found
                Test:           TestSyncFullWithBridge
```

### Relevant log output

_No response_

### Notes

Seems that this only shows up on Mac, also only shows up with keyring backend `test`

### PR DESCRIPTION
**Pull request by renaynay**
This PR fixes flakeyness observed mainly on macOS when running integration tests (swamp tests) that had to use non-empty blocks.

Resolves #1669

### CODE REVIEW

**Comment by Wondertan** [comment]
My turn to cry: 😭

Jokes aside, do you now why it fixes the issue? Isn't 1 sec enough and why 2?

**Comment by renaynay** [comment]
@Wondertan don't worry about timeout too much, we now send err through chan even if it's nil, so we will know when the server actually starts instead of waiting 5 seconds even if operation already returned successfully

File: core/testing_grpc.go [hunk 1]
```diff
@@ -109,6 +109,5 @@ func startGRPCServer(
 			log.Error("serving GRPC: ", err)
 		}
 	}()
-
```
Comment (Wondertan) [inline core/testing_grpc.go:112]: remove

**Review by Wondertan — APPROVED** [review]
Much better

**Review by Wondertan — DISMISSED** [review]

**Review by distractedm1nd — DISMISSED** [review]

**Review by vgonkivs — APPROVED** [review]

**Review by derrandz — COMMENTED** [review]
On first run by running `make test-swamp`, we had only one test case fail instead of all 4 (_ref the issue_)

```

--- FAIL: TestFullReconstructFromLights (1.61s)
    reconstruct_test.go:170: 
                Error Trace:    /Users/derrandz/Lab/Celestia/celestia-node-fork/nodebuilder/tests/reconstruct_test.go:170
                Error:          Received unexpected error:
                                rpc error: code = NotFound desc = account celestia1259ljlnng56hekhdc5mue4j7mey9gfv6dpltuk not found
                Test:           TestFullReconstructFromLights
```

Upon running this test case in isolation, the issue didn't pop up.

Screen Shot 2023-02-02 at 14 39 13

So this double confirms that the issue comes from some sort of flakiness.

**Review by Wondertan — APPROVED** [review]

**Pull request closed without merging**